src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2348,6 +2348,40 @@ void CodeGen::genX86BaseIntrinsic(GenTreeHWIntrinsic* node, insOpts instOptions)
 
     switch (intrinsicId)
     {
+        case NI_X86Base_BigMul:
+        case NI_X86Base_X64_BigMul:
+        {
+            assert(node->GetOperandCount() == 2);
+            assert(instOptions == INS_OPTS_NONE);
+            assert(!node->Op(1)->isContained());
+
+            // SIMD base type is from signature and can distinguish signed and unsigned
+            var_types   targetType = node->GetSimdBaseType();
+            GenTree*    regOp      = node->Op(1);
+            GenTree*    rmOp       = node->Op(2);
+            instruction ins        = HWIntrinsicInfo::lookupIns(intrinsicId, targetType, compiler);
+
+            emitAttr attr = emitTypeSize(targetType);
+            emitter* emit = GetEmitter();
+
+            if (rmOp->isUsedFromReg() && rmOp->GetRegNum() == REG_EAX)
+            {
+                std::swap(rmOp, regOp);
```

Comment:
Could you please explain this swap to me?

Reply:
In case op2 (rmop) is already present in RAX we use op2 as implicit operand. 
Otherwise we would overwrite that value.

I should probably add a comment similar to the one I added for codeGenMulHi 
https://github.com/dotnet/runtime/blob/51a412389aecbde55d8183ef94d5e9d09ab4732a/src/coreclr/jit/codegenxarch.cpp#L868-L871


Do you want me to add the comment to this PR or https://github.com/dotnet/runtime/pull/115966 ?

Reply:
I'm fine with either option, I think the PR is not too big to split it?

Reply:
> I think the PR is not too big to split it?

I've updated both PRs. 
The 2 reasons to open this as a separate PR was to 

1.  Show that it works for the non-AVX2 path and passes all tests
2. For the mulx path I am a bit unsure about the best way to [handle the allocation of the RDX register the MULX code produced worse code than MUL for at least one method](https://github.com/dotnet/runtime/pull/115966#discussion_r2155554140) (you can expand comment to se asm code) . Either approach is better than what's in main but I wanted to avoid adding a "mulx" optimization that gives worse code on average